src/modules/sched-simple/sched.c

```diff
@@ -330,6 +330,45 @@ static void cancel_cb (flux_t *h,
     }
 }
 
+/* Job manager indicates there is a priority change to a job.  If a
+ * matching job found in queue, update the priority and reorder queue
+ * as necessary.
+ */
+void prioritize_cb (flux_t *h, flux_msg_handler_t *mh,
+                    const flux_msg_t *msg, void *arg)
+{
+    struct simple_sched *ss = arg;
+    json_t *jobs;
+    size_t index;
+    json_t *arr;
+
+    if (flux_request_unpack (msg, NULL, "{s:o}", "jobs", &jobs) < 0) {
+        flux_log_error (h, "%s: flux_request_unpack", __FUNCTION__);
+        return;
+    }
+
+    json_array_foreach (jobs, index, arr) {
+        flux_jobid_t id;
+        unsigned int priority;
+        json_int_t tmp;
+        struct jobreq *job;
+
+        if (json_unpack (arr, "[I,I]", &id, &tmp) < 0) {
+            flux_log_error (h, "%s: invalid prioritize entry sent",
```

Comment:
I don't know that we want to do it here, but there's an opportunity for an optimization:  If there are many priority updates, it may be faster to defer the reorder call and just sort the queue after all the priorities have been updated.

Reply:
> I would use int64_t here instead of json_int_t for consistency.
> Also may as well change jobreq->priority to an int64_t.

So this bleeds into issue https://github.com/flux-framework/flux-core/issues/3450

Do we want to use `int64_t` *everywhere*, including for API parameters?  B/c right now I'm working under the assumption that users of the `libschedutil` API will (eventually) want to use the native type `unsigned int`.

Right now we are "hand parsing" the `sched.prioritize` RPC from `job-manager`, but it will eventually be taken care of via a function `libschedutil`.

Reply:
Oh yeah, `jobreq->priority` should probably track the libschedutil API, and IMHO there it makes sense to go with an `unsigned int` (or `uint32_t`).